backends/vllm/src/model_executor.rs

```diff
@@ -296,19 +353,55 @@ where
             sender.send(output).ok();
         }
 
+        #[cfg(feature = "nccl")]
+        while let Some(command) = self.receiver.blocking_recv() {
```

Comment:
refactor this block, it's too much similar code to the above. If you put the non nccl sender into an option as well, it will be even easier.

Reply:
Done